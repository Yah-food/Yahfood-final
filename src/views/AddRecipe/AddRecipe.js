import React from 'react';
import { useState } from 'react';
import RecipeForm from '../../components/RecipeForm/RecipeForm';
import { createNewRecipe } from '../../services/users';

export default function AddRecipe() {
  const [newRecipe, setNewRecipe] = useState({});

  const updateRecipeState = (key, value) => {
    newRecipe[key] = value;
    setNewRecipe({ ...newRecipe });
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    try {
      await createNewRecipe(newRecipe);
      alert('New recipe has been added!');
    } catch {
      alert('Error, try again');
    }
    // window.location.replace('/profile');
  };

  return (
    <div>
      <RecipeForm
        {...newRecipe}
        handleSubmit={handleSubmit}
        updateRecipeState={updateRecipeState}
      />
    </div>
  );
}
