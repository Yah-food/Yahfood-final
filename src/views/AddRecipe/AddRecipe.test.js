import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import AddRecipe from './AddRecipe';

test('renders add recipe form page', () => {
  const container = render(
    <MemoryRouter>
      <AddRecipe />
    </MemoryRouter>
  );
  expect(container).toMatchSnapshot();
});
