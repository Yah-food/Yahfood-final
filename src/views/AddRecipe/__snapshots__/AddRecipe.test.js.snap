// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders add recipe form page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="pt-2"
        >
          <a
            class="mx-5"
            href="/profile"
          >
            <button>
              Back
            </button>
          </a>
        </div>
        <div
          class="flex justify-center font-sans text-5xl opacity-70 py-10"
        >
          <h1>
            yahfood!
          </h1>
        </div>
        <div>
          <form
            class="py-20"
          >
            <label
              class="flex justify-center items-center flex-col"
            >
              Title:
              <input
                class=" border-solid rounded-lg border-2 w-40"
                name="title"
                type="text"
                value=""
              />
            </label>
            <label
              class="flex justify-center items-center flex-col"
            >
              Ingredients:
              <textarea
                class=" border-solid rounded-lg border-2 w-40"
                name="description"
                type="text-area"
              />
            </label>
            <label
              class="flex justify-center items-center flex-col"
            >
              Instructions:
              <textarea
                class=" border-solid rounded-lg border-2 w-40"
                name="instructions"
                type="text-area"
              />
            </label>
            <label
              class="flex justify-center items-center flex-col"
            >
              Image url:
              <input
                class=" border-solid rounded-lg border-2 w-40"
                name="images"
                type="text"
                value=""
              />
            </label>
            <div
              class="flex justify-center mx-auto my-5 w-64 items-center border-2 border-lime-700/50 rounded-lg bg-lime-700/50 "
            >
              <button>
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="pt-2"
      >
        <a
          class="mx-5"
          href="/profile"
        >
          <button>
            Back
          </button>
        </a>
      </div>
      <div
        class="flex justify-center font-sans text-5xl opacity-70 py-10"
      >
        <h1>
          yahfood!
        </h1>
      </div>
      <div>
        <form
          class="py-20"
        >
          <label
            class="flex justify-center items-center flex-col"
          >
            Title:
            <input
              class=" border-solid rounded-lg border-2 w-40"
              name="title"
              type="text"
              value=""
            />
          </label>
          <label
            class="flex justify-center items-center flex-col"
          >
            Ingredients:
            <textarea
              class=" border-solid rounded-lg border-2 w-40"
              name="description"
              type="text-area"
            />
          </label>
          <label
            class="flex justify-center items-center flex-col"
          >
            Instructions:
            <textarea
              class=" border-solid rounded-lg border-2 w-40"
              name="instructions"
              type="text-area"
            />
          </label>
          <label
            class="flex justify-center items-center flex-col"
          >
            Image url:
            <input
              class=" border-solid rounded-lg border-2 w-40"
              name="images"
              type="text"
              value=""
            />
          </label>
          <div
            class="flex justify-center mx-auto my-5 w-64 items-center border-2 border-lime-700/50 rounded-lg bg-lime-700/50 "
          >
            <button>
              Submit
            </button>
          </div>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
